pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java

```diff
@@ -707,13 +726,17 @@ private void testPulsarSourceLocalRun(String jarFilePathUrl) throws Exception {
         }
     }
 
+    @Test(timeOut = 20000, groups = "builtin")
+    public void testPulsarSourceStatsBuiltin() throws Exception {
+        testPulsarSourceLocalRun(String.format("%s://data-generator", Utils.BUILTIN));
+    }
 
-    @Test
+    @Test(timeOut = 20000)
```

Comment:
I am not sure we need these timeouts,
if the test breaks then it is not likely that other tests will run.

Reply:
The individual timeout is in place so we don't just block for 1.5-2 hour which is the absolute test timeout and consume resources for that time.

Reply:
We centrally set 300 seconds timeouts for all the tests that don't have a timeout